osprey/strategies.py

```diff
@@ -168,61 +172,45 @@ def _hyperopt_fmin_random_kwarg(random):
-            raise ValueError("method must be one of 'epi', 'kriging', "
-                             "'constant_liar'")
-
-        if lie_method not in ('constant_liar_min', 'constant_liar_max',
-                              'constant_liar_mean'):
-            raise ValueError("method lie_method be one of 'constant_liar_min',"
-                             " 'constant_liar_max', 'constant_liar_mean'")
-
-        if self.url:
-            self._use_local_moe = False
-        else:
-            try:
-                # check if it's importable
-                from moe.views.rest.gp_next_points_epi import GpNextPointsEpi
-                # force flake8 not to complain about unused import
-                bool(GpNextPointsEpi)
-                self._use_local_moe = True
-            except ImportError as e:
-                print(e, file=sys.stderr)
-                msg = ('with strategy = "moe", either "url" parameter must be '
-                       'set to point to an external MOE REST API, or you must '
-                       'have a local copy of MOE installed and importable. '
-                       'See the MOE documentation at '
-                       'http://yelp.github.io/MOE/ for details')
-                raise RuntimeError(msg)
-
-    def suggest(self, history, searchspace):
-        """Suggest params to maximize an objective function based on the
-        function evaluation history using a Gaussian Process Expected
-        Parallel Improvement (GPEI) algorithm, as implemented by MOE [1].
-
-        .. [1] http://yelp.github.io/MOE/index.html
-        """
-        request = self._build_request(history, searchspace)
-
-        if self._use_local_moe:
-            results = self._call_moe_locally(request)
-        else:
-            results = self._call_moe_rest_api(request)
-
-        return self._build_response(results, searchspace)
-
-    def _build_request(self, history, searchspace):
-        points_sampled = []
-        points_being_sampled = []
+class GP(BaseStrategy):
+    short_name = 'gp'
+
+    def __init__(self, kernel=None, max_feval=5E4, max_iter=1E5):
+        self.kernel = kernel
+        self.max_feval = max_feval
+        self.max_iter = max_iter
+        self.model = None
+        self.n_dims = None
+
+    def _set_kernel(self):
+        self.kernel = (Matern52(self.n_dims, ARD=True) +
+                       White(self.n_dims) +
+                       Bias(self.n_dims))
+
+    def _fit_model(self, X, Y):
+        model = GPRegression(X, Y, self.kernel)
+        model.optimize(messages=False, max_f_eval=self.max_feval)
+        self.model = model
+
+    def _optimize(self, init=None, uncertainty=False):
+        if not init:
+            init = self._init()
+
+        def z(x):
+            y = x.copy().reshape(-1, self.n_dims)
+            s, v = self.model.predict(y)
+            s[(x < 0.)*(x > 1.)] = -1
+            v[(x < 0.)*(x > 1.)] = -1
+            if uncertainty:
+                return -v.flatten()
```

Comment:
Is this right? I dont know the api, but don't we want to add the mean and std dev or something to minimize the upper edge of the confidence interval or something?